File: Strings/1_10/7Solution.py
from collections import Counter
def greatestLetter(s):
        count=Counter(s)
            
        for ascii_value in reversed(range(65, 65+26)):
            u=chr(ascii_value)
            l=u.lower()
            if count[u] and count[l]:
                return u
        
        return ''

s=input()
print(greatestLetter(s))

File: Strings/1_10/7Solution.java
class Solution {
    public String greatestLetter(String s) {
        int[] lower=new int[26];
        int[] upper=new int[26];
        for(char c:s.toCharArray()){
            if(c>='A'&&c<='Z'){
                upper[c-'A']++;
            }else{
                lower[c-'a']++;
            }
        }
        for(int i=25;i>=0;i--){
            if(upper[i]>0&&lower[i]>0){
                return String.valueOf((char)(i+'A'));
            }
        }
        return "";
    }
}
